br/pkg/registry/registration.go

```diff
@@ -195,6 +226,17 @@ func (r *Registry) executeInTransaction(ctx context.Context, fn func(context.Con
 // ResumeOrCreateRegistration first looks for an existing registration with the given parameters.
 // If found and paused, it tries to resume it. Otherwise, it creates a new registration.
 func (r *Registry) ResumeOrCreateRegistration(ctx context.Context, info RegistrationInfo) (uint64, error) {
+	// check for tasks with same parameters except restoredTS (common PiTR auto-detection issue)
+	if err := r.checkForAutoRestoredTSConflict(ctx, info); err != nil {
+		return 0, err
+	}
+
+	// clean up stale tasks first
+	if err := r.CleanupStaleRunningTasks(ctx); err != nil {
```

Comment:
can we make a seperated goroutine to clean up the stale task? otherwise, the main routine could be blocked to more than 5 minutes

Reply:
discussed offline. This part of logic has been removed. However, the new resume logic also needs to block for 5 mins but that's necessary since it needs to decide whether to reuse the restore id or not.